Fixes SR-642: Code completion does not instantiate generic arguments of a type wrapped in an optional.

ahoppen opened the pull request.

nkcsgexi commented:
Could you add a test?

gribozavr commented on lib/IDE/CodeCompletion.cpp:
```diff
@@ -1708,7 +1708,7 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
       if (ContextTy) {
         Type MaybeNominalType = ExprType->getRValueInstanceType();
-        if (ContextTy->getAnyNominal() == MaybeNominalType->getAnyNominal() &&
+        if (ContextTy->getDesugaredType()->getAnyNominal() == MaybeNominalType->getDesugaredType()->getAnyNominal() &&
```
80 columns.

gribozavr commented:
... and it would be best if the test didn't use Set, but used a mock type instead.  `Set` can change in future.

ahoppen commented:
I added a test case and added a line break so that the lines are only 80-ish columns. Thanks for the hint with the mock type, wouldn’t have thought about that

NachoSoto commented on test/IDE/complete_generic_optional.swift:
```diff
@@ -0,0 +1,16 @@
+let x: Foo<Bar>? = Foo<Bar>()
+x.#^FOO_OPTIONAL_1^#
+// FOO_OPTIONAL_1: Begin completions, 4 items
+// FOO_OPTIONAL_1-DAG: ecl[InstanceMethod]/CurrNominal/Erase[1]: ?.myFunction({#(foobar): Bar#})[#Void#]; name=myFunction(foobar: Bar)
```
This is awesome :+1:

gribozavr replied:
*D*ecl?

gribozavr commented on test/IDE/complete_generic_optional.swift:
```diff
@@ -0,0 +1,16 @@
+
+struct Foo<T> {
+	func myFunction(foobar: T) {
+	}
```
Please use two spaces for indentation.

ahoppen commented:
Sorry, don't know how these changes happened between my tests and and the commit. Hope everything is fine now ;-)

tkremenek commented:
@swift-ci Please test
